target/linux/ipq807x/image/generic.mk

```diff
@@ -36,6 +36,33 @@ define Device/edimax_cax1800
 endef
 TARGET_DEVICES += edimax_cax1800
 
+define Device/netgear_sxk80
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_VENDOR := Netgear
+	DEVICE_DTS_CONFIG := config@hk01
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	SOC := ipq8074
+	IMAGE/sysupgrade.bin/squashfs := append-rootfs | pad-to 64k | \
```

Comment:
Why is this custom squashfs recipe needed?
Whats the point in 64k padding on a 128k native NAND?

Reply:
I must admit that I still haven't tested the actual installation yet as I am not sure about the right way to go forward. What would you suggest? It is necessary to change the bootcmd in uboot anyways (as it's doing secureboot image verification), my approach for the install procedure would be this:
1. Enable telnet using http://<ip address>/debug.htm
2. Telnet into the device
3. Modify u-boot environment to set bootcmd to "tftpboot; bootm"
4. Reboot and have it load the initramfs
5. Modify the bootcmd again to set it to something (this is depending on how I build the image I guess? But what's the "best" way here?)
6. Use sysupgrade to reboot
7. OpenWRT should be installed

So what recipe should I use in this case? Just remove that line completely and see if it boots?

Reply:
I could also use a DNI image and use the normal bootdni instead of using secureboot I guess. Not sure what is "better" though?

Reply:
I was referring to the build recipe, as you have 128k NAND while squashfs is being overridden to 64k

Reply:
I understand, but what recipe I choose depends on how I change the u-boot bootcmd, right? And as I need to change the bootcmd anyways to disable secureboot, what's the "best" recipe/bootcmd to use? If I decide to use dni then I need to build a dni image for example, but is that the best option or is something more simple better? Removing this line entirely for example? But what's the correct bootcmd for that case then?

Reply:
Is the stock FW using DNI images?
If not, there is no point in custom stuff anyway, and the regular UBI image should work

Reply:
They use their secureboot dni image, yes. But we can't generate those anyways as we can't sign them.

Reply:
Then just the regular images should work as you need to change the bootcmd anyway